tests/fv/flannel_migration_test.go

```diff
@@ -0,0 +1,381 @@
+	}
+
+	stopController := func() {
+		migrationController.Stop()
+	}
+
+	BeforeEach(func() {
+		// Run etcd.
+		etcd = testutils.RunEtcd()
+
+		// Run apiserver.
+		apiserver = testutils.RunK8sApiserver(etcd.IP)
+
+		// Write out a kubeconfig file
+		kconfigfile, err = ioutil.TempFile("", "ginkgo-migrationcontroller")
+		Expect(err).NotTo(HaveOccurred())
+
+		data := fmt.Sprintf(testutils.KubeconfigTemplate, apiserver.IP)
+		_, err = kconfigfile.Write([]byte(data))
+		Expect(err).NotTo(HaveOccurred())
+
+		k8sClient, err = testutils.GetK8sClient(kconfigfile.Name())
+		Expect(err).NotTo(HaveOccurred())
+
+		// Wait for the apiserver to be available.
+		Eventually(func() error {
+			_, err := k8sClient.CoreV1().Namespaces().List(metav1.ListOptions{})
+			return err
+		}, 30*time.Second, 1*time.Second).Should(BeNil())
+
+		// Apply the necessary CRDs.
+		out, err := apiserver.ExecOutput("kubectl", "apply", "-f", "crds.yaml")
+		Expect(err).NotTo(HaveOccurred(), fmt.Sprintf("Failed to apply CRDs: %s", out))
+
+		// Make a Calico client and backend client.
+		type accessor interface {
+			Backend() backend.Client
+		}
+		calicoClient = testutils.GetCalicoClient(apiconfig.Kubernetes, "", kconfigfile.Name())
+		bc = calicoClient.(accessor).Backend()
+
+		// Run controller manager.
+		controllerManager = testutils.RunK8sControllerManager(apiserver.IP)
+
+		// Initialise a new Flannel cluster
+		flannelCluster = testutils.NewFlannelCluster(k8sClient, "192.168.0.0/16")
+
+		// Setup cluster
+		flannelCluster.AddFlannelDaemonset(flannelDs)
+		flannelCluster.AddCalicoDaemonset("calico-node")
+		flannelCluster.AddDefaultCalicoConfigMap()
+
+		// Add master and controller nodes.
+		flannelCluster.AddFlannelNode(controllerNodeName, "192.168.1.0/24", "vxlan", "8e:74:9a:30:24:01", "172.16.0.1", emptyLabel, false)
+		flannelCluster.AddFlannelNode(masterNodeName, "192.168.2.0/24", "vxlan", "8e:74:9a:30:24:02", "172.16.0.2", emptyLabel, true)
+		flannelCluster.AddFlannelNode("node-2", "192.168.3.0/24", "vxlan", "8e:74:9a:30:24:03", "172.16.0.3", emptyLabel, false)
+	})
+
+	AfterEach(func() {
+		flannelCluster.Reset()
+		os.Remove(kconfigfile.Name())
+		controllerManager.Stop()
+		apiserver.Stop()
+		etcd.Stop()
+	})
+
+	Context("Basic FV tests", func() {
+		BeforeEach(func() {
+			startController()
+		})
+		AfterEach(func() {
+			stopController()
+		})
+
+		It("should initialize the datastore at start-of-day", func() {
+			var info *api.ClusterInformation
+			Eventually(func() *api.ClusterInformation {
+				info, _ = calicoClient.ClusterInformation().Get(context.Background(), "default", options.GetOptions{})
+				return info
+			}).ShouldNot(BeNil())
+
+			Expect(info.Spec.ClusterGUID).To(MatchRegexp("^[a-f0-9]{32}$"))
+			Expect(info.Spec.ClusterType).To(Equal("k8s,kdd"))
+			Expect(*info.Spec.DatastoreReady).To(BeTrue())
+		})
+
+		Context("Healthcheck FV tests", func() {
+			It("should pass health check", func() {
```

Comment:
Huh, I didn't realize we had health checks in the migration controller. I can't seem to find that code - could you point me at it?

Reply:
Migration controller does not has it's own health checks but it use standard checks for api server connection. 
https://github.com/projectcalico/kube-controllers/blob/master/cmd/kube-controllers/main.go#L183